data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -1,59 +1,101 @@
 // Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
-use data_pipeline::trace_exporter::TraceExporter;
-use data_pipeline::trace_exporter::TraceExporterBuilder;
+use data_pipeline::trace_exporter::{
+    ResponseCallback, TraceExporter, TraceExporterInputFormat, TraceExporterOutputFormat,
+};
 use ddcommon_ffi::{
     slice::{AsBytes, ByteSlice},
-    CharSlice,
+    CharSlice, MaybeError,
 };
-use std::ffi::c_char;
+use std::{ffi::c_char, ptr::NonNull};
 
+/// Create a new TraceExporter instance.
+///
+/// # Arguments
+///
+/// * `out_handle` - The handle to write the TraceExporter instance in.
+/// * `url` - The URL of the Datadog Agent to communicate with.
+/// * `tracer_version` - The version of the client library.
+/// * `language` - The language of the client library.
+/// * `language_version` - The version of the language of the client library.
+/// * `language_interpreter` - The interpreter of the language of the client library.
+/// * `input_format` - The input format of the traces. Setting this to Proxy will send the trace
+///   data to the Datadog Agent as is.
+/// * `output_format` - The output format of the traces to send to the Datadog Agent. If using the
+///   Proxy input format, this should be set to format if the trace data that will be passed through
+///   as is.
+/// * `agent_response_callback` - The callback into the client library that the TraceExporter uses
```

Comment:
[Ignore this comment, speaking out loud] Longer term, I'd love to actually make this more functional, i.e registering for sampling rates rather than an agent response.